File: BlazorAddressifySolution/BlazorAddressify/wwwroot/exampleJsInterop.js
// This is a JavaScript module that is loaded on demand. It can export any number of
// functions, and may import other JavaScript modules if required.

export function showPrompt(message) {
  return prompt(message, 'Type anything here');
}


export function getAddressPosition() {
    const elem = document.getElementById("ba-address");
    const rect = elem.getBoundingClientRect();

    for (let key in rect) {
        if (typeof rect[key] !== 'function') {
            var con = `${key} : ${rect[key]}`;
            console.log(con);
        }
    }

    //return rect["top"] + ";" + rect["left"] + ";" + rect["height"];

    return {
        top: rect["top"],
        left: rect["left"],
        height: rect["height"],
        width: rect["width"]
        };
}

File: BlazorAddressifySolution/BlazorAddressify/ExampleJsInterop.cs
using Microsoft.JSInterop;
using System;
using System.Text.Json;
using System.Threading.Tasks;

namespace BlazorAddressify
{
    // This class provides an example of how JavaScript functionality can be wrapped
    // in a .NET class for easy consumption. The associated JavaScript module is
    // loaded on demand when first needed.
    //
    // This class can be registered as scoped DI service and then injected into Blazor
    // components for use.

    public class ExampleJsInterop : IAsyncDisposable
    {
        private readonly Lazy<Task<IJSObjectReference>> moduleTask;

        public ExampleJsInterop(IJSRuntime jsRuntime)
        {
            moduleTask = new(() => jsRuntime.InvokeAsync<IJSObjectReference>(
               "import", "./_content/BlazorAddressify/exampleJsInterop.js").AsTask());
        }

        public async ValueTask<string> Prompt(string message)
        {
            var module = await moduleTask.Value;
            return await module.InvokeAsync<string>("showPrompt", message);
        }

        public async ValueTask<AddressPosition> GetAddressPosition()
        {
            var module = await moduleTask.Value;
            var vals = await module.InvokeAsync<AddressPosition>("getAddressPosition");
            return vals;
        }

        public class AddressPosition 
        { 
            public double top { get; set; }
            public double left { get; set; }
            public double height { get; set; }
            public double width { get; set; }
        }
        public async ValueTask DisposeAsync()
        {
            if (moduleTask.IsValueCreated)
            {
                var module = await moduleTask.Value;
                await module.DisposeAsync();
            }
        }
    }
}
